src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -2693,6 +2693,29 @@ class TypeSpecTest {
     }
   }
 
+  @Test
+  fun testDelegateIfaceWithOtherParamTypeName() {
+    val type = TypeSpec.classBuilder("EntityBuilder")
+        .primaryConstructor(FunSpec.constructorBuilder()
+            .addParameter(ParameterSpec.builder("argBuilder",
+                ParameterizedTypeName.get(ClassName.bestGuess("Payload"),
+                    ClassName.bestGuess("EntityBuilder"),
+                    ClassName.bestGuess("Entity")))
+                .defaultValue("Payload.create()")
+                .build())
+            .build())
+        .addSuperinterface(ParameterizedTypeName.get(ClassName.bestGuess("TypeBuilder"),
+            ClassName.bestGuess("EntityBuilder"),
+            ClassName.bestGuess("Entity")), "argBuilder")
+        .build()
+
+    assertThat(toString(type)).isEqualTo("""
+          |package com.squareup.tacos
+          |
+          |class EntityBuilder(argBuilder: Payload<EntityBuilder, Entity> = Payload.create()) : TypeBuilder<EntityBuilder, Entity> by argBuilder
```

Comment:
Don't know exactly where this is handled, but carriage returns aren't inserted here for this long class signature (it normally formats them correctly when adding regular supertypes, if I recall correctly)

Reply:
Normally this is done by adding a `%W` modifier into the `CodeBlock`, that will trigger the `CodeWriter` to emit either a whitespace, or new line if the current one is too long. Can we do this:

```kotlin
if (superTypes.isNotEmpty()) {
  codeWriter.emitCode(superTypes.joinToCode(prefix = " : ", separator = ",%W"))
}
```

Can you please check if that works for you?

Reply:
Seems it's the other way around, I used:

`codeWriter.emitCode(superTypes.joinToCode(separator = ",%W", prefix = " : "))`

I checked the [style guide](https://kotlinlang.org/docs/reference/coding-conventions.html#class-header-formatting), and it says that the supertypes list/constructor invocation should be on the same line as the closing parentheses, so I think technically the right way to format this case is by breaks on parameter:

     class EntityBuilder(
          argBuilder: Payload<EntityBuilder, Entity> = Payload.create()
    ) : TypeBuilder<EntityBuilder, Entity> by argBuilder

It looks like it'd need context/lookaround when formatting the entire type with constructor, name, etc. to do that esp. since prefixing all superinterfaces with a break is allowed if there's a superclass ctor invocation, so for now I just added the separator as shown above which formats this example from the tests correctly:

        class Guac(somethingElse: String) : RandomIface by Factory.byStrategy(FacadeImpl),
            Runnable by Runnable({ Logger.debug("Hello world") })

Reply:
Looks good! In any case, I don't feel like proper styling and wrapping is that important, since this is generated code. Wrapping long lines doesn't hurt though.